apps/prairielearn/elements/pl-order-blocks/dag_checker.py

```diff
@@ -204,7 +206,9 @@ def lcs_partial_credit(
     """
     graph = dag_to_nx(depends_graph, group_belonging)
     trans_clos = nx.transitive_closure(graph)
-    submission_no_distractors = [node for node in submission if node in depends_graph]
+    submission_no_distractors = [
+        node for node in submission if node in depends_graph and node is not None
+    ]
```

Comment:
can the "is not None" part of this be automatically inferred from the fact that `depends_graph` is typed as `Mapping[str, list[str]]` ? 

What happens if you don't change the list comprehension at all, but just try to annotated `submission_no_distractors` as `list[str]`?

Reply:
pyright is not smart enough to figure that out, I tried the annotation as well before I did the `is not None`.

Reply:
that's disappointing but ok I guess